dev/src/cli/cli_create.ts

```diff
@@ -0,0 +1,262 @@
+    process.exit(0);
+  }
+
+  if (!overwriteFolderResponse) {
+    console.error(`Agent directory ${agentDir} already exists.`);
+    process.exit(0);
+  }
+  await removeFolder(agentDir);
+  await createFolder(agentDir);
+}
+
+async function generateFiles(options: AgentCreationOptions) {
+  const agentPath = path.join(dirname, options.agentName, `agent.${options.language}`);
+  const agentDir = path.dirname(agentPath);
+  const dotEnvPath = path.join(agentDir, '.env');
+
+  await saveToFile(agentPath, AGENT_TEMPLATE(options.model || 'gemini-2.5-flash'));
+
+  const lines = [];
+  if (options.apiKey) {
+    lines.push(`GOOGLE_API_KEY=${options.apiKey}`);
+    lines.push(`GOOGLE_GENAI_USE_VERTEXAI=0`);
+  }
+  if (options.project) {
+    lines.push(`GOOGLE_CLOUD_PROJECT=${options.project}`);
+  }
+  if (options.region) {
+    lines.push(`GOOGLE_CLOUD_LOCATION=${options.region}`);
+  }
+  if (options.region && options.project) {
+    lines.push(`GOOGLE_GENAI_USE_VERTEXAI=1`);
+  }
+  await saveToFile(dotEnvPath, lines.join('\n'));
+}
+
+export async function createAgent(options: AgentCreationOptions) {
+  const agentDir = path.join(dirname, options.agentName);
+  await generateAgentFolder(agentDir);
+
+  if (!options.model) {
+    const model: symbol | string = (await select(
+      {
+        message: 'Choose a model for the root agent',
+        options: [
+          { label: 'gemini-2.5-flash', value: 'gemini-2.5-flash' },
+          { label: 'gemini-2.5-pro', value: 'gemini-2.5-pro' },
+          { label: 'gemini-3-pro-preview', value: 'gemini-3-pro-preview' },
+          { label: 'gemini-3-flash-preview', value: 'gemini-3-flash-preview' },
+        ],
+      },
+    ));
+    if(isCancel(model)) {
+      process.exit(0);
+    }
+    options.model = model;
+  }
+  
+
+  if(options.language !== "js" && options.language !== "ts") {
+    const language = (await select({
+        message: 'Choose a language for the agent',
+        options: [
+          { label: 'JavaScript', value: 'js' },
+          { label: 'TypeScript', value: 'ts' },
+        ],
+      },
+    ));
+    if(isCancel(language)) {
+      process.exit(0);
+    }
+    options.language = language;
+  }
+
+  if (!options.apiKey && !options.project) {
+    const backend: symbol | string = (await select({
+        message: 'Choose a backend',
+        options: [
+          { label: 'Google AI', value: 'googleai' },
+          { label: 'Vertex AI', value: 'vertex' },
+        ],
+      },
+    ));
+    if(isCancel(backend)) {
+      process.exit(0);
+    }
+
+    if (backend === 'vertex') {
+      const projectResponse: string = (await text(
+        {
+          message: 'Enter the Google Cloud Project ID',
+          initialValue: await getGcpProject(),
+        },
+      )).toString();
+      if(isCancel(projectResponse)) {
+        process.exit(0);
+      }
+      options.project = projectResponse;
+
+      const regionResponse: symbol | string = (await text({
```

Comment:
why the `symbol`? Isn't it just a string?

Reply:
If you press CTRL+C while the script is running we want to capture the cancellation which comes back as a symbol in the lib we are using.